lib/ansible/utils/display.py

```diff
@@ -297,6 +310,13 @@ def __init__(self, verbosity=0):
         except Exception as ex:
             self.warning(f"failed to patch stdout/stderr for fork-safety: {ex}")
 
+        codecs.register_error('_replacing_warning_handler', _replacing_warning_handler(self))
```

Comment:
Since we're already tied to an instance of `display` anyway (and we're doing this inside its init), could we not just `register_error` a bound method on the current display singleton sans the decoration magic, or at least use `functools.partial` to pass in the current `display` instead of the RYO decorator? I had to look sideways at it for a minute to figure out what was going on...